test/performance/logging.cc

```diff
@@ -0,0 +1,205 @@
+  {
+    bucketsWithEmpty[idx] = 0;
+  }
+
+  for (auto value : all_measurements)
+  {
+    ++bucketsWithEmpty[value];
+    ++buckets[value];
+  }
+
+  /*
+  std::cout << "\n\n Microsecond bucket measurement" << std::endl;
+  for (const auto ms_bucket : buckets) {
+     std::cout << ms_bucket.first << "\t, " << ms_bucket.second << std::endl;
+  }
+  std::cout << "\n\n";
+  */
+
+  std::ostringstream oss;
+  // Save to xcel and google doc parsable format. with empty buckets
+  oss << "\n\n Microsecond bucket measurement with zero buckets till max"
+      << std::endl;
+  for (const auto ms_bucket : bucketsWithEmpty)
+  {
+    oss << ms_bucket.first << "\t, " << ms_bucket.second << std::endl;
+  }
+  WriteToFile(result_filename, oss.str());
+  std::cout << "Worst Case Latency, max value: " << kMaxValue << std::endl;
+  std::cout << "microsecond bucket result is in file: " << result_filename
+            << std::endl;
+}
+}  // namespace
+
+void run(size_t number_of_threads)
+{
+  g_counter = 0;
+  ignition::common::Console::SetVerbosity(4);
+  std::vector<std::thread> threads(number_of_threads);
+  std::map<size_t, std::vector<uint64_t>> threads_result;
+
+  for (size_t idx = 0; idx < number_of_threads; ++idx)
+  {
+    // reserve to 1 million for all the result
+    // it's a test so  let's not care about the wasted space
+    threads_result[idx].reserve(g_iterations);
+  }
+
+  // int queue_size = 1048576; // 2 ^ 20
+  // int queue_size = 524288;  // 2 ^ 19
+  // spdlog::set_async_mode(queue_size); // default size is 1048576
+  auto filename_result = std::to_string(number_of_threads) + ".result.csv";
+  std::ostringstream oss;
+  oss << "Using " << number_of_threads;
+  oss << " to log in total " << g_iterations << " log entries to "
+      << filename_result << std::endl;
+  WriteToFile(filename_result, oss.str());
+
+  auto start_time_application_total =
+  std::chrono::high_resolution_clock::now();
+  for (uint64_t idx = 0; idx < number_of_threads; ++idx)
+  {
+    threads[idx] = std::thread(MeasurePeakDuringLogWrites, idx,
+                               std::ref(threads_result[idx]));
+  }
+  for (size_t idx = 0; idx < number_of_threads; ++idx)
+  {
+    threads[idx].join();
+  }
+  auto stop_time_application_total = std::chrono::high_resolution_clock::now();
+
+  uint64_t total_time_in_us =
+      std::chrono::duration_cast<std::chrono::microseconds>(
+          stop_time_application_total - start_time_application_total)
+          .count();
+
+  PrintStats(filename_result, threads_result, total_time_in_us);
+  SaveResultToBucketFile(filename_result, threads_result);
+}
+
+class LoggingTest:
+      public ::testing::TestWithParam<std::size_t>
+{
+};
+
+TEST_P(LoggingTest, RunThreads)
+{
+  run(GetParam());
+  SUCCEED();
+}
+
+INSTANTIATE_TEST_SUITE_P(LoggingTest, LoggingTest,
+                         ::testing::Values(1, 2, 4, 8, 16, 32));
+
+/////////////////////////////////////////////////
+int main(int argc, char **argv)
```

Comment:
Are we keeping this test? If so maybe we can add a comment saying that the test passes if it runs without any segfaults or something to that effect